### PR TITLE
Use the AbstractContextManager from SpiNNUtils

### PR DESCRIPTION
**Pull request by dkfellows**
Also shut up some stupid warnings from rig; we're not fixing them anyway

### CODE REVIEW

**Comment by dkfellows** [comment]
See https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon/pull/680

**Comment by dkfellows** [comment]
I've also made it so that we don't depend on rig (and can't use the command line option for booting) unless the user installs this package with `pip install "spalloc[boot]"`. I don't plan to document this more widely; we don't need it for our own purposes.

File: requirements.txt [hunk 1]
```diff
@@ -16,4 +16,5 @@
 six>=1.8.0
 appdirs
 enum-compat
+SpiNNUtilities >= 1!5.1.1, < 1!6.0.0
```
Comment (Christian-B) [inline requirements.txt:19]: While I don't think this is an issue better check with @lplana that he does not mind a new dependency for spalloc!

File: setup.py [hunk 2]
```diff
@@ -67,6 +68,12 @@
             "spalloc-where-is = spalloc.scripts.where_is:main",
         ],
     },
+    # Booting directly needs rig; not recommended! Use SpiNNMan instead, as
+    # that has an up-to-date boot image pre-built
+    extras_require={
+        'boot': [
+            'rig',
```
Comment (Christian-B) [inline setup.py:75]: This looks kike an add of a rig requirement not a removal!

File: requirements.txt (hunk 1 above)
Comment (lplana) [reply requirements.txt:19]: > While I don't think this is an issue better check with @lplana that he does not mind a new dependency for spalloc!

No objection from me.

Comment (dkfellows) [reply requirements.txt:19]: This isn't for the server, this is for the client.

File: setup.py (hunk 2 above)
Comment (dkfellows) [reply setup.py:75]: It's describing what was there already; we were already (in one of the command line tools) looking to see if it was there and enabling extra options if it was.

**Pull request merged**